### PR TITLE
C125-146: Legend size

### PR DESCRIPTION
**Pull request by dsuren1**
## Description
Added possibility to set width and height for legends

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [ ] Bugfix
 - [x] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue
https://github.com/geosolutions-it/austrocontrol-C125/issues/146
#5176

**What is the current behavior?**
User can set custom legend options i.e width and height 

**What is the new behavior?**
User did not have an option to set custom width and height from the UI

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

## Other useful information

### CODE REVIEW

File: web/client/components/TOC/fragments/settings/General.jsx [hunk 1]
```diff
@@ -187,12 +187,63 @@ class General extends React.Component {
...
+                            <FormGroup>
+                                <ControlLabel><Message msgId="layerProperties.legendOptions.legendWidth" /></ControlLabel>
+                                <FormControl
+                                    defaultValue={this.props.element.legendOptions && this.props.element.legendOptions.legendWidth || ''}
+                                    key="legendWidth"
+                                    type="number"
+                                    onKeyPress={(e)=> {
+                                        const number = parseInt(e.target.value + e.key, 10);
+                                        const result = Number.isInteger(number);
+                                        !result || e.key === "." ? e.preventDefault() : result && number > 1000 && e.preventDefault();
```
Comment (offtherailz) [inline web/client/components/TOC/fragments/settings/General.jsx:204]: Negative numbers should not be allowed, as well as too big numbers.

File: web/client/translations/data.pt-PT.json [hunk 2]
```diff
@@ -101,6 +101,11 @@
                 "right": "Right",
                 "bottom": "Bottom",
                 "top": "Top"
+            },
+            "legendOptions": {
```
Comment (offtherailz) [inline web/client/translations/data.pt-PT.json:105]: Mandatory translations are :
en, it, es, fr, de. You can avoid to update the other files.

File: web/client/translations/data.it-IT.json [hunk 3]
```diff
@@ -106,6 +106,11 @@
                 "right": "Destra",
                 "bottom": "Sotto",
                 "top": "Sopra"
+            },
+            "legendOptions": {
+              "title": "Leggenda",
```
Comment (offtherailz) [inline web/client/translations/data.it-IT.json:111]:
```suggestion
              "title": "Legenda",
```

File: web/client/components/TOC/fragments/WMSLegend.jsx [hunk 4]
```diff
@@ -49,6 +68,14 @@ class WMSLegend extends React.Component {
     canShow = (node) => {
         return node.visibility || !this.props.showOnlyIfVisible;
     };
+
+    useLegendOptions = () =>{
```
Comment (offtherailz) [inline web/client/components/TOC/fragments/WMSLegend.jsx:72]: Why > 12? I it is a limit, I think that also the UI should be explicit on this limitation (e.g. input doesn't allow these values)

File: web/client/components/TOC/fragments/settings/General.jsx (hunk 1 above)
Comment (dsuren1) [reply web/client/components/TOC/fragments/settings/General.jsx:204]: Yes, this will not allow negative numbers and also the max user can feed is 1000

**Pull request merged**